src/kernel/klib.c

```diff
@@ -101,8 +134,11 @@ closure_function(3, 1, status, load_klib_complete,
     klib_debug("   loading klib @ %R, resolving relocations\n", kl->load_range);
     elf_apply_relocs(b, where);
 
+    klib_debug("   ingesting elf symbols\n");
+    add_elf_syms(kl->elf, kl->load_range.start);
```

Comment:
Why is this being moved here? This code can run multiple times for a single klib (e.g. the first time it may fail with KLIB_MISSING_DEP, and then after another klib is loaded, the second time it succeeds); having this code run only when KLIB_INIT_OK or KLIB_INIT_IN_PROGRESS ensures elf symbols are ingested only once per klib.

Reply:
Prior to doing the tail copy, I initially had klib_elf_map map the last page of the data section and simply zero the non-page-aligned portion of the bss section. This would cause the elf symbol ingest to later fail, as I think this zeroing was affecting either the symtab or stringtab sections. I moved the elf ingest prior to the elf loading for this reason, but missed that it could be invoked multiple times.

Since I subsequently changed all the non-faulting elf map handlers do the tail copy, this is no longer necessary. I have now reverted this change.